### DISCUSSION TITLE
Latch variable of EGS_Particle Struct

### DISCUSSION THREAD

**Discussion by jstask0** [nrc-cnrc/EGSnrc#1059] [Questions] [answered]
I was looking through the documentation of EGS_Particle, and it appears that the latch variable is an ordinary int. For BEAMnrc, I'm familiar with the latch being bits (either 1 or 0). Could I set the latch of an EGS_Particle to be some other value like 2 or 3? Or is it like the latch in BEAMnrc where it's 0 or not 0?

**Comment by ftessier** [comment, marked as answer]
The meaning of `latch` is determined by the application (BEAMnrc, egs_chamber, etc.). It is always just an integer, but the application might interpret it as a numerical value (1, 2, 3, ...) or a bit field (correspondingly: `0x01`, `0x10`, `0x11`, ...). In BEAMnrc, it is interpreted as a bit field, where each bit in the integer is turned on (1) or off (0) according to the latch bit associations in the geometry.

So again, `latch` is always a plain old integer, but in the application you can set its value and interpret it as you wish.

**Comment by mchamberland** [comment]
I always thought @randlet handled latch really neatly in egs_brachy. [See here for some ideas. ](https://github.com/clrp-code/egs_brachy/blob/main/egs_brachy/latch.h)
